values-tls.yaml

```diff
@@ -0,0 +1,172 @@
+# Default values for infinispan-helm-charts.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+images:
+  # [USER] The container images for server pods.
+  server: quay.io/infinispan/server:14.0
+  initContainer: registry.access.redhat.com/ubi8-micro
+
+deploy:
+  # [USER] Specify the number of nodes in the cluster.
+  replicas: 1
+
+  clusterDomain: cluster.local
+
+  container:
+    extraJvmOpts: ""
+    libraries: ""
+    # [USER] Define custom environment variables using standard K8s format
+    # env:
+    #  - name: STANDARD_KEY
+    #    value: standard value
+    #  - name: CONFIG_MAP_KEY
+    #    valueFrom:
+    #      configMapKeyRef:
+    #        name: special-config
+    #        key: special.how
+    #  - name: SECRET_KEY
+    #    valueFrom:
+    #      secretKeyRef:
+    #        name: special-secret
+    #        key: special.how
+    env:
+    storage:
+      size: 1Gi
+      storageClassName: ""
+      # [USER] Set `ephemeral: true` to delete all persisted data when clusters shut down or restart.
+      ephemeral: false
+    resources:
+      # [USER] Specify the CPU limit and the memory limit for each pod.
+      limits:
+        cpu: 500m
+        memory: 512Mi
+      # [USER] Specify the maximum CPU requests and the maximum memory requests for each pod.
+      requests:
+        cpu: 500m
+        memory: 512Mi
+
+  security:
+    secretName: ""
+    batch: ""
+  ssl:
+   secretName: "tls-secret"
+  expose:
+    # [USER] Specify `type: ""` to disable network access to clusters.
+    type: Route
+    nodePort: 0
+    host: ""
+    annotations: []
+
+  monitoring:
+    enabled: false
+
+  logging:
+    categories:
+      # [USER] Specify the FQN of a package from which you want to collect logs.
+      - category: com.arjuna
+        # [USER] Specify the level of log messages.
+        level: warn
+      # No need to warn about not being able to TLS/SSL handshake
+      - category: io.netty.handler.ssl.ApplicationProtocolNegotiationHandler
+        level: error
+
+  makeDataDirWritable: false
+
+  nameOverride: ""
+
+  resourceLabels: []
+
+  podLabels: []
+
+  svcLabels: []
+
+  tolerations: []
+
+  nodeAffinity: {}
+
+  nodeSelector: {}
+
+  infinispan:
+    cacheContainer:
+      # [USER] Add cache, template, and counter configuration.
+      name: default
+      # [USER] Specify `security: null` to disable security authorization.
+      security:
+        authorization: {}
+      transport:
+        cluster: ${infinispan.cluster.name:cluster}
+        node-name: ${infinispan.node.name:}
+        stack: kubernetes
+        urn:infinispan:server:14.0:securityRealm: default
```

Comment:
With the current configuration we're utilising the same security realm for both the cluster transport and the user endpoint. In the Infinispan server guide we advise users not to do this: https://infinispan.org/docs/stable/titles/server/server.html#securing-cluster-transport-tls-identities_secure-cluster-transport

I think we need to provide a mechanism that allows users to configure TLS at the endpoint and/or transport layer.

How about exposing the following:

```yaml
  ssl:
   endpointSecretName: "endpoint-secret" # Mount at /etc/encrypt/endpoint
   transportSecretName: "transport-secret" # Mount at /etc/encrypt/transport
```